[29기 정지민] 상세페이지

rindalica opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
-상세페이지 러프 레이아웃 구현
-scss 대신 styled component로 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
상세 페이지의 전체 페이지 레이아웃을 러프하게 구현
지도 API 가져오기만 함.

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
>받아오는 데이터 내에서 
>>사진 리스트의 길이가 0일 경우
>>사진 리스트 길이가 0이 아닌 경우
>>>사진 리스트 길이가 4장인 경우
>>>사진 리스트 길이가 2장인 경우
총 2번의 조건부 랜더링 진행

>데이터 받아올 때 객체 구조분해할당
>스크롤과 무관하게 위치가 고정되어 있는 예약 컴포넌트 : position : sticky
>예약하기 버튼 클릭시 선택한 일정 및 인원수 데이터 백엔드로 전달

> react date picker 
>> 달력 라이브러리

> react recoil
>> 전역 상태 관리 라이브러리

<br />

## :: 추가 구현 사항
상세페이지 내 NAVBAR
내용 및 댓글 <더보기> 버튼 구현
호스트 소개
후기 별점, 평가 기준

201steve commented on src/pages/Detail/Detail.js:
```diff
@@ -0,0 +1,22 @@
+import DetailContents from './DetailContents/DetailContents';
+import DetailReviewMap from './DetailReviewMap/DetailReviewMap';
+
+const DetailPage = styled.div`
```
* 컴포넌트 파일에 진입하면, `styled componenet`보다 `jsx`가 먼저 보일 수 있게 위치를 조정 해 주세요.

rindalica replied:
> * 컴포넌트 파일에 진입하면, `styled componenet`보다 `jsx`가 먼저 보일 수 있게 위치를 조정 해 주세요.

수정했습니다!

201steve commented on src/pages/Detail/DetailContents/DetailHouseCategories.js:
```diff
@@ -0,0 +1,76 @@
+function DetailHouseCategories() {
+  return (
+    <DetailHouseCategoriesStyle>
+      {/* 데이터 받아오면 map으로 구현 예정 */}
```
* 그럼 여기 주석엔 TODO 혹은 FIXME로 적어주면 더더욱 좋겠네요

rindalica replied:
> * 그럼 여기 주석엔 TODO 혹은 FIXME로 적어주면 더더욱 좋겠네요

수정했어요!

201steve commented on src/pages/Detail/DetailContents/DetailHouseIntro.js:
```diff
@@ -0,0 +1,42 @@
+        *수건, 어메니티 등의 개인 위생 관련 용품은 구비되어 있지 않습니다.
+        <br />
+        (길건너에 마트와 다이소가 있습니다.)
+        {/* 추가구현
```
* 여기도 마찬가지로 TODO or FIXME

rindalica replied:
> * 여기도 마찬가지로 TODO or FIXME

여기도 수정했습니다!

201steve commented on src/pages/Detail/DetailContents/DetailHouseCategories.js:
```diff
@@ -0,0 +1,76 @@
+      {/* 데이터 받아오면 map으로 구현 예정 */}
+      <DetailHouseCategoriesTitle>숙소 포함사항</DetailHouseCategoriesTitle>
+      <DetailHouseCategoriesBox>
+        <DetailHouseCategory>
+          <DetailHouseCategoryIcon>icon</DetailHouseCategoryIcon>
+          <DetailHouseCategoryTitle>탈주로</DetailHouseCategoryTitle>
+        </DetailHouseCategory>
+        <DetailHouseCategory>
+          <DetailHouseCategoryIcon>icon</DetailHouseCategoryIcon>
+          <DetailHouseCategoryTitle>함정</DetailHouseCategoryTitle>
+        </DetailHouseCategory>
+        <DetailHouseCategory>
+          <DetailHouseCategoryIcon>icon</DetailHouseCategoryIcon>
+          <DetailHouseCategoryTitle>병원</DetailHouseCategoryTitle>
+        </DetailHouseCategory>
+        <DetailHouseCategory>
+          <DetailHouseCategoryIcon>icon</DetailHouseCategoryIcon>
+          <DetailHouseCategoryTitle>좀비</DetailHouseCategoryTitle>
+        </DetailHouseCategory>
+        <DetailHouseCategory>
+          <DetailHouseCategoryIcon>icon</DetailHouseCategoryIcon>
+          <DetailHouseCategoryTitle>도깨비</DetailHouseCategoryTitle>
+        </DetailHouseCategory>
```
* 반복되는 UI는 .map으로 구현 해 주세요

rindalica replied:
> * 반복되는 UI는 .map으로 구현 해 주세요

나중에 데이터 받으면 map으로 구현하겠습니다. 주석으로 추가로 적어놨습니다.

201steve commented on src/pages/Detail/DetailReviewMap/DetailReviews.js:
```diff
@@ -0,0 +1,104 @@
+            <DetailReviewUserImg>이미지</DetailReviewUserImg>
+            <DetailReviewUserInfo>
+              <DetailReviewUserName>닉네임</DetailReviewUserName>
+              <DetailReviewDate>2022년 2월</DetailReviewDate>
+            </DetailReviewUserInfo>
+          </DetailReviewUser>
+          <DetailReviewContent>
+            도망가자 어디든 가야 할 것만 같아 넌 금방이라도 울 것 같아 괜찮아
+            우리 가자 걱정은 잠시 내려놓고 대신 가볍게 짐을 챙기자 실컷 웃고
+            다시 돌아오자 거기서는 우리 아무 생각말자
+          </DetailReviewContent>
+        </DetailReview>
+        <DetailReview>
+          <DetailReviewUser>
+            <DetailReviewUserImg>이미지</DetailReviewUserImg>
+            <DetailReviewUserInfo>
+              <DetailReviewUserName>닉네임</DetailReviewUserName>
+              <DetailReviewDate>2022년 2월</DetailReviewDate>
+            </DetailReviewUserInfo>
+          </DetailReviewUser>
+          <DetailReviewContent>
+            도망가자 어디든 가야 할 것만 같아 넌 금방이라도 울 것 같아 괜찮아
+            우리 가자 걱정은 잠시 내려놓고 대신 가볍게 짐을 챙기자 실컷 웃고
+            다시 돌아오자 거기서는 우리 아무 생각말자
+          </DetailReviewContent>
+        </DetailReview>
+        <DetailReview>
+          <DetailReviewUser>
+            <DetailReviewUserImg>이미지</DetailReviewUserImg>
+            <DetailReviewUserInfo>
+              <DetailReviewUserName>닉네임</DetailReviewUserName>
+              <DetailReviewDate>2022년 2월</DetailReviewDate>
+            </DetailReviewUserInfo>
+          </DetailReviewUser>
+          <DetailReviewContent>
+            도망가자 어디든 가야 할 것만 같아 넌 금방이라도 울 것 같아 괜찮아
+            우리 가자 걱정은 잠시 내려놓고 대신 가볍게 짐을 챙기자 실컷 웃고
+            다시 돌아오자 거기서는 우리 아무 생각말자
+          </DetailReviewContent>
+        </DetailReview>
```
* 여기도 반복되는 ui네요. .map으로 구현 해 주세요

rindalica replied:
> * 여기도 반복되는 ui네요. .map으로 구현 해 주세요

여기는 추가구현 항목이라.. 나중에 추가 구현시 수정하겠습니다!

201steve reviewed (commented):
수고하셨습니다!
리뷰 수정해서 반영 해 주세요!

rindalica commented on src/pages/Detail/DetailContents/DetailHouseReservation.js:
```diff
@@ -0,0 +1,34 @@
+  background-color: white;
+  border: 1px solid rgb(221, 221, 221);
+  border-radius: 12px;
+  box-shadow: rgb(0 0 0 / 12%) 0px 6px 16px;
```
> * theme에 같은 컬러 속성이 변수로 지정 되있네요
> * theme에서 꺼내 쓰는거로 적용 해 주세요

수정했습니다~!

rindalica commented on src/pages/Detail/DetailContents/DetailHouseCalendar.js:
```diff
@@ -0,0 +1,93 @@
+import Calendars from '../../../components/Calendars/Calendars';
+
+function DetailHouseCalendar() {
+  const [startDate, setStartDate] = useRecoilState(AtomStartDate);
```
데이터에 키값이 x

rindalica commented on src/pages/Detail/DetailTop/FiveImgs.js:
```diff
@@ -0,0 +1,26 @@
+  return (
+    <DetailSmallImageBox>
+      {list.map(img => {
+        return <DetailSmallImage alt="smallImages" src={img} />;
```
데이터에 키값이 x

201steve commented:
사이더 이슈 해결 해 주세요!

201steve commented on src/pages/Detail/Detail.js:
```diff
@@ -0,0 +1,55 @@
+      {details && (
+        <>
+          <DetailTop
+            name={details.result.name}
```
* 구조분해할당 적용 할 수 있습니다.

rindalica replied:
수정했습니다~!

201steve commented on src/pages/Detail/DetailContents/DetailHostIntro.js:
```diff
@@ -0,0 +1,46 @@
+}
+
+const DetailHostIntroStyle = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
* display:flex, justify-content, align-items는 전역에서 자주 반복해서 쓰는 속성입니다.
* 속성값은 레이아웃에따라 가변적으로 쓰이기 때문에
* theme에서 mix-in으로 선언해서 쓰는거로 수정 해 주세요

rindalica replied:
수정했습니다~

201steve commented on src/pages/Detail/DetailContents/DetailHouseCalendar.js:
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import styled from 'styled-components';
+import { useRecoilState } from 'recoil';
+// import DatePicker from 'react-datepicker';
+// import 'react-datepicker/dist/react-datepicker.css';
```
불필요한 주석은 삭제해주세요!
* 코드가 주석처리된 채로 남아있다면 필요한 상황에 제대로 사용되거나 삭제되는 일은 거의 없고 결국 계속 남아있게 됩니다.
* 그러다보면 이제 작성한 사람이 아닌 다른 개발자는 필요한 코드라서 주석처리한게 아닐까? 라는 생각으로 지우지 못하고
* 결국 그 누구도 왜 있는지는 모르면서 계속 남아있는 불필요한 레거시가 되어버립니다!
* 따라서 반드시 필요한 경우가 아니면 코드를 주석처리하기보다는 삭제해주시고 만약 다시 보고싶다면 커밋 이력에 잘 남겨뒀다가 원하는 시점에 다시 보면 됩니다!

rindalica replied:
이 부분은 나중에 수정할 예정입니다!

201steve commented on src/pages/Detail/DetailContents/DetailHouseCategories.js:
```diff
@@ -0,0 +1,59 @@
+    <DetailHouseCategoriesStyle>
+      <DetailHouseCategoriesTitle>흉가 포함사항</DetailHouseCategoriesTitle>
+      <DetailHouseCategoriesBox>
+        <DetailHouseCategory>
+          <DetailHouseCategoryTitle>
+            흉가 유형 - {houseType}
+          </DetailHouseCategoryTitle>
+        </DetailHouseCategory>
```
* 이 UI가 반복되고있네요 반복되는 UI는 .map으로 구현 해 주세요

rindalica replied:
데이터로 받아오는 부분 외에 적어놓은 부분들이 다 달라서 map 으로 구현하기 어려울 것 같아요 ㅠㅠ

201steve replied:
리팩토링 할때 참고!

201steve commented on src/pages/Detail/DetailMap/DetailMap.js:
```diff
@@ -0,0 +1,62 @@
+        {city} · {country}
+      </DetailMapDetail>
+      <LoadScript
+        googleMapsApiKey="AIzaSyBNOa7yNsc8oZTDIDjoNKb5aKzNCsDog3k"
```
- apikey 같은 민감한 정보는 유출되면 경제적 손실에까지 이를 수 있습니다.
- 따라서, github같은 열려있는 공간에 노출되지 않도록 환경변수 파일(.env)로 관리 해 줘야 합니다.
- 환경변수로 관리하는 정보들은 local로 팀에 공유 해 주세요.

rindalica replied:
이 부분 수정했습니다!

201steve reviewed (commented):
CRA 초기파일이 추가되있습니다. 
필요없는 파일은 삭제 해 주시고, sider 이슈 해결 해 주세요

ghost commented:
Sider has detected **1** warning on analyzing the commit d42e73a.

- ⚠️ The number of issues (41) exceeded the limit of comments. Please visit [Sider](https://sider.review/gh/repos/459037917/pulls/2) to review the issues.

If the errors persist even after [retrying](https://sider.review/gh/repos/459037917/pulls/2#tools), the following actions may resolve them:
- Update tool dependencies
- Configure failed tools
- [Configure `sider.yml`](https://sider.review/gh/repos/459037917/pulls/2#config)
- [Turn off unused tools](https://sider.review/gh/repos/459037917/settings)

If you still have problems, feel free to ask us via chat. 💬

---

You can [turn off such notifications](https://sider.review/gh/repos/459037917/settings#notifications) if unnecessary.

201steve reviewed (commented):
sider 다시 풀었어요 확인 해주세요!

201steve commented on src/pages/Detail/DetailContents/DetailHouseReservation.js:
```diff
@@ -0,0 +1,256 @@
+`;
+
+const DetailHouwReservationModal = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
```
* theme 적용 해 주세요

201steve commented on src/pages/Detail/DetailMap/DetailMap.js:
```diff
@@ -0,0 +1,63 @@
+    lat: Number(latitude),
+    lng: Number(longitude),
+  };
+  // const googleMapKey = process.env.REACT_APP_GOOGLE_MAP_API_KEY;
+  // console.log(googleMapKey);
```
주석 필요없다면 삭제 해 주세요!

201steve commented on src/pages/Detail/DetailMap/DetailMap.js:
```diff
@@ -0,0 +1,63 @@
+        {city} · {country}
+      </DetailMapDetail>
+      <LoadScript
+        googleMapsApiKey="AIzaSyAL_UUynGfb-zlmk_dpkn9kqYwf1H3JeQE"
```
* apikey는 github에 올라오면 안됩니다
* 환경변수에서 가져다가 쓸 수 있게 해주세요

201steve reviewed (commented):
고지가 얼마 안남았습니다

rindalica committed:
ADD : Rough Layout
- refactor
- Add : date picker install
- refactor : PR 반영
- refactor : 달력 분리 전
- refactor
- Delete : calendar
- docs
- refactor
- refactor
- 백엔드 통신 확인
- Add : google map key
- refactor -02/24
- 달력 수정
- image key
- theme mixin 추가
- theme 수정
- google map api key
- theme 수정

201steve reviewed (approved):
LGTM👍🏻
